docs/reference-guide/query-modules/implement-custom-query-modules/custom-query-module-example.md

```diff
@@ -441,6 +585,68 @@ As mentioned before, no exceptions should leave your module. If you are writing
 the module in a language that throws them, use exception handlers
 in `mgp_init_module` and `mgp_shutdown_module` as well.
 
+
+### Batched query procedures
+
+Similar to batched query procedures in Python, you can add batched query procedures in C.
+
+Batched procedures need 3 functions, one for each of batching, initialization, and cleanup.
+
+```c
+static void batch(const struct mgp_list *args, const struct mgp_graph *graph,
+                      struct mgp_result *result, struct mgp_memory *memory) {
+  ...
+}
+
+static void init(const struct mgp_list *args, const struct mgp_graph *graph,
+                 struct mgp_memory *memory) {
+  ...
+}
+
+static void cleanup() {
+  ...
+}
+```
+
+
+The `batch` function receives the list of arguments (`args`) passed in the
+query. The parameter `result` is used to fill in the resulting records of the
+procedure. Parameters `graph` and `memory` are context parameters of the
+procedure, and they are used in some parts of the provided C API. `batch`
+needs to return at some point empty `result` which signals that the `batch` procedure is done with execution and `cleanup` can be called. `init` doesn't receive `result` as it is only used for initialization. `init` function will receive same arguments which are registered and passed to `batch` function.
+Memgraph ensures to call `init` before the `batch` function, and `cleanup` at the end. The user invokes directly the `batch` function through OpenCypher.
+The passed in `memory` argument is only alive throughout the execution of
+`mgp_init_module`, so you must not allocate any global resources with it.  Consequently, you may want to reset any global state or release global resources
+in the `cleanup` function. 
+
+For more information on what exactly is possible with C API, take a look at the
+`mg_procedure.h` file or the [C API reference
+guide](/reference-guide/query-modules/implement-custom-query-modules/api/c-api.md).
+
+The following line contains the `mgp_init_module` function that registers procedures
+that can be invoked through Cypher. Even though the example has only one
+`procedure`, you can register multiple different procedures in a single module.
+
+Batch procedures are invoked using the `CALL <module>.<batch_procedure> ...` syntax. The
+`<module-name>` will correspond to the name of the shared library. Since we
+compile our example into `example.so`, then the module is called `example`.
+As mentioned, Memgraph ensures to call `init` before  `<batch_procedure>` and `cleanup` once `<batch_procedure>` signals end with an empty result.
```

Comment:
```suggestion
Batch procedures are invoked using the `CALL <module>.<batch_procedure> ...` syntax. The
`<module-name>` will correspond to the name of the shared library. Since the example is complied into `example.so`, the module is called `example`.
As mentioned, Memgraph ensures to call `init` before  `<batch_procedure>` and `cleanup` once `<batch_procedure>` signals end with an empty result.
```